api/client/client.go

```diff
@@ -370,6 +373,29 @@ func (c *Client) Dialer() ContextDialer {
 	return c.dialer
 }
 
+// WaitForConnectionReady waits for the client's grpc connection finish dialing,
+// returning true if it becomes ready and returning false if the ctx is canceled
+// or the connection enters a closed state. This can be used alongside the
+// DialInBackground client config option to wait until background dialing has completed.
+func (c *Client) WaitForConnectionReady(ctx context.Context) bool {
+	for {
+		switch state := c.conn.GetState(); state {
```

Comment:
Nit: I think you can just do `switch state := c.conn.GetState() {`, or is `; state` part needed?

Reply:
The `; state` part is needed here, unlike for type switches.